src/amber/server/server.cr

```diff
@@ -67,9 +67,17 @@ module Amber
         exit
       end
 
-      logger.info "Server started in #{colorize(Amber.env, :yellow)}."
-      logger.info colorize("Startup Time #{Time.now - time}\n\n", :white)
-      server.listen(settings.port_reuse)
+      loop do
+        begin
+          logger.info "Server started in #{colorize(Amber.env, :yellow)}."
+          logger.info colorize("Startup Time #{Time.now - time}\n\n", :white)
+          server.listen(settings.port_reuse)
+          exit
+        rescue e : Errno
+          logger.error e.message
+          logger.info "Restarting server..."
```

Comment:
I think it would be helpful to admins if this type of event could be easily hooked with an initializer. Perhaps something like this:

```crystal
# ...
rescue e : Errno
  server_crash(e)
  logger.error e.message
  logger.info "Restarting server..."
end

def server_crash(exception : Errno)
  # noop
end
```

Reply:
@robacarp Feel free to submit a PR on that. I'm super swamped right now and want to get this merged in since it's causing issues.